crates/worker/src/partition/mod.rs

```diff
@@ -415,7 +406,7 @@ where
                     self.on_rpc(rpc, &mut partition_store).await;
                 }
                 _ = status_update_timer.tick() => {
-                    self.status.last_archived_log_lsn = *self.last_snapshot_lsn_watch.1.borrow();
+                    self.status.last_archived_log_lsn = *self.archived_lsn_receiver.borrow();
```

Comment:
Could this be set by the `PartitionProcessorManager`? That way we wouldn't need this extra watcher from the PPM to the `PartitionProcessor`.